sdk/go/pulumi/resource.go

```diff
@@ -217,6 +219,39 @@ type resourceOptions struct {
 	Version string
 }
 
+func awaitResourceInput(ctx context.Context, ri ResourceInput) (Resource, []Resource, error) {
+	result, known, _, deps, err := ri.ToResourceOutput().await(ctx)
+
+	if !known {
+		return nil, nil, fmt.Errorf("Encountered unknown ResourceInput, this is currently not supported")
```

Comment:
Not yet obvious to me of whether we should try to support it somehow already.  I need figure out an example representative of the use case.